plugins/src/main/java/com/google/fhir/gateway/plugin/OpenSRPHelper.java

```diff
@@ -0,0 +1,532 @@
+package com.google.fhir.gateway.plugin;
+
+import static org.smartregister.utils.Constants.EMPTY_STRING;
+
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.gclient.ReferenceClientParam;
+import com.google.fhir.gateway.ProxyConstants;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.r4.model.BaseResource;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.CareTeam;
+import org.hl7.fhir.r4.model.Enumerations;
+import org.hl7.fhir.r4.model.Group;
+import org.hl7.fhir.r4.model.Identifier;
+import org.hl7.fhir.r4.model.Location;
+import org.hl7.fhir.r4.model.Organization;
+import org.hl7.fhir.r4.model.OrganizationAffiliation;
+import org.hl7.fhir.r4.model.Practitioner;
+import org.hl7.fhir.r4.model.PractitionerRole;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smartregister.model.location.LocationHierarchy;
+import org.smartregister.model.location.ParentChildrenMap;
+import org.smartregister.model.practitioner.FhirPractitionerDetails;
+import org.smartregister.model.practitioner.PractitionerDetails;
+import org.smartregister.utils.Constants;
+import org.springframework.lang.Nullable;
+
+public class OpenSRPHelper {
```

Comment:
can we rename this class with something more semantic? Like `RelationshipManagementHelper`

Reply:
The idea was that it would be generic to contain other helper methods for extended functionality. Right now it supports the multi-practitioner details endpoint so maybe for now we can re-name it to something that alludes to that.

Reply:
The idea was that it would be generic to contain other helper methods for extended functionality. Right now it only supports the multi-practitioner details endpoint so maybe for now we can re-name it to something that alludes to that.

Reply:
Renamed it from `OpenSRPHelper` to `PractitionerDetailsEndpointHelper`